pkg/controller.v1/pytorch/pytorchjob_controller.go

```diff
@@ -347,32 +351,40 @@ func (r *PyTorchJobReconciler) UpdateJobStatus(job interface{},
 					msg := fmt.Sprintf("PyTorchJob %s is running.", pytorchjob.Name)
 					err := commonutil.UpdateJobConditions(jobStatus, commonv1.JobRunning, commonutil.JobRunningReason, msg)
 					if err != nil {
-						commonutil.LoggerForJob(pytorchjob).Infof("Append job condition error: %v", err)
+						logger.Infof("Append job condition error: %v", err)
 						return err
 					}
 				}
 				// when master is succeed, the job is finished.
 				if expected == 0 {
 					msg := fmt.Sprintf("PyTorchJob %s is successfully completed.", pytorchjob.Name)
-					logrus.Info(msg)
+					logger.Info(msg)
 					r.Recorder.Event(pytorchjob, corev1.EventTypeNormal, commonutil.JobSucceededReason, msg)
 					if jobStatus.CompletionTime == nil {
 						now := metav1.Now()
 						jobStatus.CompletionTime = &now
 					}
 					err := commonutil.UpdateJobConditions(jobStatus, commonv1.JobSucceeded, commonutil.JobSucceededReason, msg)
 					if err != nil {
-						commonutil.LoggerForJob(pytorchjob).Infof("Append job condition error: %v", err)
+						logger.Infof("Append job condition error: %v", err)
 						return err
 					}
 					trainingoperatorcommon.SuccessfulJobsCounterInc(pytorchjob.Namespace, pytorchv1.FrameworkName)
 					return nil
 				}
 			}
 		} else {
+
 			if rtype == pytorchv1.PyTorchReplicaTypeWorker {
-				// TODO(gaocegege): Support SuccessPolicy
-				if expected == 0 {
+				worker0Completed, err := r.IsWorker0Completed(pytorchjob, replicas)
+				if err != nil {
+					logger.Warnf("check if worker 0 completed error %v", err)
+					return err
+				}
+				// Leave a succeeded condition for the following two cases:
+				// 1. If default success policy is used and worker 0 has completed.
+				// 2. If `SuccessPolicyAllWorkers` success policy is used and all workers are succeeded.
+				if expected == 0 || (worker0Completed && *pytorchjob.Spec.SuccessPolicy != pytorchv1.SuccessPolicyAllWorkers) {
```

Comment:
L387 might take a bit of time to understand. Maybe it's because the order of the two conditions are not arranged as the comment lists.

Reply:
The comments and conditions are copied from TF. I will refine it soon.